### PR TITLE
Poppler qt6

### PR DESCRIPTION
**Pull request by sunderme**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>? -> no, but neither does any other formula I have seen, including foo.rb

-----
This pull-request stems from the discussion https://github.com/Homebrew/discussions/discussions/1771

### CODE REVIEW

File: Formula/poppler.rb
Comment (SMillerDev) [inline Formula/poppler.rb:79]: This might make future version bumps easier
```suggestion
        "#{lib}/libpoppler-qt#{Formula["qt"].version.major}.dylib",
```

**Comment by gromgit** [comment]
```
pdal:
  * pdal contains conflicting version recursive dependencies:
      qt, qt@5
    View these with `brew deps --tree pdal`.
```
Probably means you need to change `pdal` to `depends_on "poppler-qt5"`...
```
poppler-qt5:
  * Formula pdftohtml should also have a conflict declared with poppler-qt5
  * Formula pdf2image should also have a conflict declared with poppler-qt5
  * Formula xpdf should also have a conflict declared with poppler-qt5
```
...and add `poppler-qt5` to the`conflicts_with` stanzas on those three formulae.

File: Formula/poppler.rb
Comment (sunderme) [reply Formula/poppler.rb:79]: it will be a while until qt7 comes up ...

**Comment by gromgit** [comment]
> ...and add `poppler-qt5` to the`conflicts_with` stanzas on those three formulae.

For this, you should wait for #81249 to be merged, then `brew update` and `git rebase origin/master` in your PR branch _before_ doing any changes. I made changes to both `pdftohtml` and `xpdf` in preparation for my own attempt at bumping `poppler` to Qt6, to avoid odd audit errors down the line.

File: Formula/poppler-qt5.rb
Comment (carlocab) [inline Formula/poppler-qt5.rb:19]:
```suggestion
```
New formulae shouldn't have bottle blocks.

File: Formula/poppler-qt5.rb
Comment (carlocab) [inline Formula/poppler-qt5.rb:41]: We should probably make this keg-only, instead of declaring multiple conflicts.

**Review by carlocab — COMMENTED** [review]
Thanks for getting this started, @sunderme. I have a few suggestions. When you make them, please amend your commits so that:
1. You only have two commits, one for `poppler` and another for `poppler-qt5`; and,
2. Each commit message starts with the name of the formula being modified (e.g. `poppler: switch to qt6` and `poppler-qt5 21.07.0 (new formula)`).

If you don't mind, you can give @gromgit access to your PR branch, or maybe they don't mind picking up your commits and using them to open a new PR, as this one might be a bit of work.

File: Formula/poppler-qt5.rb
Comment (sunderme) [reply Formula/poppler-qt5.rb]: done

**Comment by sunderme** [comment]
maintainers can access this PR

**Comment by gromgit** [comment]
I'm not a maintainer, just an "opportunistic handyman", fixing stuff as and when I stumble across them. :laughing:

File: Formula/poppler.rb
Comment (gromgit) [reply Formula/poppler.rb:79]: True, but doing it that way means you (or anyone else) can ensure consistency when changing the "qt" dependency, just by doing a global search-and-replace (e.g. `sed -i 's/"qt"/"qt@5.7"/g Formula/poppler.rb`).

Comment (sunderme) [reply Formula/poppler.rb:79]: however there is still the cmake option. 
In this case, I am a proponent of KISS (keep it simple...)

**Comment by gromgit** [comment]
You probably want to add `revision 1` to `poppler.rb` (but keep the `bottle` block), since new bottles are needed.

File: Formula/poppler.rb
Comment (carlocab) [reply Formula/poppler.rb:79]: Yes. The suggestion here makes the formula much simpler to maintain.

**Comment by sunderme** [comment]
I am not aware how to give access to other users.
Apart from that, it is unclear what more changes are called for here.

**Comment by carlocab** [comment]
Here's a list of changes suggested above that I think are called for here:

1. Use `Formula["qt"].version.major instead of hardcoding the version.
2. Make `poppler-qt5` keg-only instead of declaring various conflicts. Failing that, the formulae that `poppler-qt5` conflicts with should declare a conflict with `poppler-qt5`, as shown by the CI errors.
3. Bump `poppler`'s revision by adding `revision 1` right after `license`.
4. `pdal` needs a `depends_on "poppler-qt5"` in place of `depends_on "poppler"`, plus a revision bump.

File: Formula/poppler.rb
Comment (carlocab) [inline Formula/poppler.rb:41]:
```suggestion
  conflicts_with "pdftohtml", "pdf2image", "xpdf",
```
You can remove this now that `poppler-qt5` is keg-only.

File: Formula/poppler-qt5.rb
Comment (carlocab) [inline Formula/poppler-qt5.rb:33]: Add a reason, please. See the other formulae that do this for reference.

File: Formula/gdal.rb
Comment (carlocab) [inline Formula/gdal.rb:46]: This also needs a revision bump.

**Review by carlocab — COMMENTED** [review]
Thanks, @sunderme. Before this is merged we'll need to fix up your commits so that there is only one commit per formula (and one formula modified per commit -- see [docs](https://docs.brew.sh/Formula-Cookbook#push)), but we can save that for later.

**Review by carlocab — COMMENTED** [review]
Please run `brew audit --strict --fix poppler-qt5` to clean up the trailing whitespace from `poppler-qt5`. Also, please amend the commit modifying `gdal` to read something like
```
gdal: change dependency to poppler-qt5
```
We put the names of formulae first in our commit messages.

**Review by carlocab — APPROVED** [review]
LGTM. Thanks for your work here, @sunderme!

**Pull request closed without merging**
